incubator/orm/index.go

```diff
@@ -27,12 +27,12 @@ type MultiKeyIndex struct {
 	indexer   indexer
 }
 
-func NewIndex(builder Indexable, prefix byte, indexer IndexerFunc) *MultiKeyIndex {
+func NewMultiKeyIndex(builder Indexable, prefix byte, indexer IndexerFunc) *MultiKeyIndex {
```

Comment:
If we go ahead with this we should also rename `IndexerFunc` to `MultiKeyIndexerFunc`